client/extensions/woocommerce/woocommerce-services/state/label-settings/reducer.js

```diff
@@ -44,6 +50,11 @@ reducers[ WOOCOMMERCE_SERVICES_LABELS_INIT_FORM ] = (
 };
 
 reducers[ WOOCOMMERCE_SERVICES_LABELS_SET_FORM_DATA_VALUE ] = ( state, { key, value } ) => {
+	let pristineData = state.pristineData;
+	if ( state.meta.pristine ) {
+		pristineData = cloneDeep( state.data );
```

Comment:
No need to `cloneDeep`, since `state` is immutable. `pristineData = state.data` is enough.